contracts/exchange/contracts/src/MixinAssetProxyDispatcher.sol

```diff
@@ -111,89 +108,19 @@ contract MixinAssetProxyDispatcher is
-                // bytes4(keccak256("transferFrom(bytes,address,address,uint256)")) = 0xa85e59e4
-                mstore(cdStart, 0xa85e59e400000000000000000000000000000000000000000000000000000000)
-
-                /////// Setup Params Area ///////
-                // Each parameter is padded to 32-bytes. The entire Params Area is 128 bytes.
-                // Notes:
-                //   1. The offset to `assetData` is the length of the Params Area (128 bytes).
-                //   2. A 20-byte mask is applied to addresses to zero-out the unused bytes.
-                mstore(add(cdStart, 4), 128)
-                mstore(add(cdStart, 36), and(from, 0xffffffffffffffffffffffffffffffffffffffff))
-                mstore(add(cdStart, 68), and(to, 0xffffffffffffffffffffffffffffffffffffffff))
-                mstore(add(cdStart, 100), amount)
-
-                /////// Setup Data Area ///////
-                // This area holds `assetData`.
-                let dataArea := add(cdStart, 132)
-                let assetDataArea := assetData
-                // solhint-disable-next-line no-empty-blocks
-                for {} lt(dataArea, cdEnd) {} {
-                    mstore(dataArea, mload(assetDataArea))
-                    dataArea := add(dataArea, 32)
-                    assetDataArea := add(assetDataArea, 32)
-                }
-
-                /////// Call `assetProxy.transferFrom` using the constructed calldata ///////
-                didSucceed := call(
-                    gas,                    // forward all gas
-                    assetProxy,             // call address of asset proxy
-                    0,                      // don't send any ETH
-                    cdStart,                // pointer to start of input
-                    sub(cdEnd, cdStart),    // length of input
-                    0,                      // don't store the returndata
-                    0                       // don't store the returndata
-                )
-
-                if iszero(didSucceed) { // Call reverted.
-                    let revertDataSize := returndatasize()
-                    // Construct a `bytes memory` type to hold the revert data
-                    // at `cdStart`.
-                    // The first 32 bytes are the length of the data.
-                    mstore(cdStart, revertDataSize)
-                    // Copy the revert data immediately after the length.
-                    returndatacopy(add(cdStart, 32), 0, revertDataSize)
-                    // We need to move the free memory pointer because we
-                    // still have solidity logic that executes after this assembly.
-                    mstore(64, add(cdStart, add(revertDataSize, 32)))
-                    revertData := cdStart
-                }
-            }
+            // Construct the calldata for the transferFrom call.
+            bytes memory proxyCalldata = abi.encodeWithSelector(
+                TRANSFER_FROM_SELECTOR,
```

Comment:
Can we use `IAssetProxy(address(0)).transferFrom.selector` here? I am making this change everywhere in a different PR.